Scripts/Source/minai_Sex.psc

```diff
@@ -1122,3 +1106,165 @@ string Function GetFactionsForActor(actor akTarget)
 
   return ret
 EndFunction
+
+String Function JoinActorArray(Actor[] actorArray) global
+  String result = ""
+  int index = 0
+  int size = actorArray.Length
+
+  ; Loop through the array using a while loop
+  while (index < size)
+      Actor currentActor = actorArray[index]
+      if (currentActor)
+          ; Append the actor's name to the result string
+          result += currentActor.GetActorBase().GetName()
+
+          ; If it's not the last actor, append a comma
+          if (index < size - 1)
+              result += ","
+          endif
+      endif
+      
+      index += 1
+  endwhile
+  
+  return result
+EndFunction
+
+; type - is db operation: start | scenechange | speedchange | end | clean
+;  start - when new thread started
+;  scenechange - when scene changed
+;  speedchange - when speed changed
+;  end - on thread end
+;  clean - clean db of all threads recorded in table
+; framework - ostim or sexlab
+; fallback - fallback description based on CBPS collision detections
+function UpdateThreadTable(string type, string framework, int ThreadID, string fallback = "")
+  actor[] actors
+  string sceneId
+  int speed = -999
+
+  ; !!!Min to review sexlab implementation
+  if(framework == "sexlab")
+    sslThreadController controller = slf.GetController(ThreadID)
+  
+    if (controller.Stage==1) 
+      LoadSexlabDescriptions()
+    EndIf
+    
+    actors = slf.GetController(ThreadID).Positions
+    sceneId = controller.Animation.FetchStage(controller.Stage)[0]
+  else
+    actors = OThread.GetActors(ThreadID)
+    sceneId = OThread.GetScene(ThreadID)
+    speed = OThread.GetSpeed(ThreadID)
+  endif
+
+  string actorsString = JoinActorArray(actors)
+
+  if(fallback == "")
+    fallback = actorsString + " have sex."
+  endif
+
+  AIAgentFunctions.logMessage("command@ExtCmdUpdateThreadsTable@{ \"type\": \""+type+"\", \"framework\": \""+framework+"\", \"threadId\": "+ThreadID+", \"actors\": \""+actorsString+"\", \"scene\": \""+sceneId+"\", \"speed\": "+speed+", \"fallback\": \""+type+"\"}@", "updateThreadsDB")
+endfunction
+
+function sexTalkClimax(actor speaker, bool chastity = false)
+  if(chastity)
+    SexTalk(speaker, "chatnf_sexclimaxchastity")
+  else
+    SexTalk(speaker, "chatnf_sexclimax")
+  endif
+endfunction
+
+function sexTalkSceneChage(actor speaker)
+  SexTalk(speaker, "chatnf_sexscenechange")
+endfunction
+
+function sexTalkSpeedChange(actor speaker)
+  SexTalk(speaker, "chatnf_sexspeedchange")
+endfunction
+
+function sexTalkOnEnd(actor speaker)
+  SexTalk(speaker, "chatnf_sexend")
+endfunction
+
+function sexTalkCollision(actor speaker, string promptToSay)
+  SexTalk(speaker, "chatnf_vr_1")
+endfunction
+
+; speaker is who actually will say llm lines, can be none if scene is player only
```

Comment:
Your comment says that none is valid, but that doesn't look to be the case looking at the below implementation

Reply:
Right - this is picking the actor that it's coming from, or the actor that it's going to? If it's "coming from", "coming from the player" is totally valid to prompt a LLM response from another actor.

Reply:
So I follow this logic:
- if player want to say something they have this opportunity by typing their line
- I'm a little confused by your concepts "coming from"/"going to". 

This function is used to select random npc which participate in scene, that's why it's excluding player.